libraries/chain/database_worker_objects.cpp

```diff
@@ -6,68 +6,24 @@
 
 namespace golos { namespace chain {
 
-    const worker_proposal_object& database::get_worker_proposal(
-        const account_name_type& author,
-        const std::string& permlink
-    ) const { try {
-        return get<worker_proposal_object, by_permlink>(std::make_tuple(author, permlink));
-    } catch (const std::out_of_range &e) {
-        GOLOS_THROW_MISSING_OBJECT("worker_proposal_object", fc::mutable_variant_object()("author",author)("permlink",permlink));
-    } FC_CAPTURE_AND_RETHROW((author)(permlink)) }
-
-    const worker_proposal_object& database::get_worker_proposal(
-        const account_name_type& author,
-        const shared_string& permlink
-    ) const { try {
-        return get<worker_proposal_object, by_permlink>(std::make_tuple(author, permlink));
+    const worker_proposal_object& database::get_worker_proposal(const comment_id_type& post) const { try {
+        return get<worker_proposal_object, by_post>(post);
     } catch (const std::out_of_range &e) {
-        GOLOS_THROW_MISSING_OBJECT("worker_proposal_object", fc::mutable_variant_object()("author",author)("permlink",permlink));
-    } FC_CAPTURE_AND_RETHROW((author)(permlink)) }
+        GOLOS_THROW_MISSING_OBJECT("worker_proposal_object", fc::mutable_variant_object()("post",post));
```

Comment:
This field is used by clients to decode error.
It is not informative for them to detect what is happened, because post_id is an internal field of the blockchain.